controller/utils.go

```diff
@@ -189,23 +191,24 @@ func replacePodsAndWait(
 				scheduler,
 				pods,
 				inRollingUpdate,
-				finishedReplace,
-				errChan,
 			)
+			// if there's an error in any of the workers cancel the operation
+			if err != nil {
+				cancel()
+				errChan <- err
+			}
 		}()
 	}
 
+	wg.Wait()
+
 	select {
 	case err = <-errChan:
-		logger.Error("operation terminated with error")
-	case <-ctx.Done():
-		logger.Debug("operation timedout/canceled")
-	case <-finishedReplace:
-		logger.Debug("all pods were successfully replaced")
+		logger.WithError(err).Debug("error replacing pods")
```

Comment:
Shouldn't this be at least a warning?

Reply:
The error is returned up the stack so the caller function can log the error, logging here just for debug